src/marketplace/AccessoriesView.js

```diff
@@ -1,11 +1,117 @@
 //@flow strict
 import * as React from 'react';
-class AccessoriesView extends React.Component<{||}> {
-    render() {    
+import {getTankComponent, verifyComponent} from '../armory/GetInventoryInfo.js';
+import Cookies from 'universal-cookie';
+
+type Props = {||}; 
+type State = {|
+	userId: string,
+|};
+let itemsForSale = [];
+class AccessoriesView extends React.Component<Props, State> {
+    constructor() {
+		super();
+		this.state={
+			//This tank id is the one the user is currently working on
+			userId: '',
+		}
+		this.getMarketSales();
+		this.getUserId();
+	}
+
+	//This gets us the user's id 
+	getUserId = async ():Promise<void> => {
+		const cookies = new Cookies();
+		const token = cookies.get('token');
+		const response = await fetch('/api/user/getUser/', {
+			method: 'GET',
+			headers: {
+				'Access-Control-Allow-Origin': '*',
+				'Content-Type': 'application/json',
+				'Access-Control-Allow-Credentials': 'true',
+				'x-auth-token': token
+			},
+		});
+		const body = await response.text();
+		const jsonObjectOfUser = JSON.parse(body);
+		this.setState({userId:jsonObjectOfUser._id});
+    };
+
+	//Gets all the sells that we are looking at 
+    getMarketSales = async ():Promise<void> => {
+		itemsForSale = [];
+		const response = await fetch('/api/marketplace/getMarketSales/', {
+			method: 'GET',
+			headers: {
+				'Access-Control-Allow-Origin': '*',
+				'Content-Type': 'application/json',
+				'Access-Control-Allow-Credentials': 'true',
+			},
+		});
+		const body = await response.text();
+		const jsonObjectOfSells = JSON.parse(body);
+		for (const sale in jsonObjectOfSells)
+		{
+			const typeOfItem = getTankComponent(verifyComponent(jsonObjectOfSells[sale].itemId));
+			if(typeOfItem === 'item')
+			{
+				let obj = {};
+				obj['name'] = jsonObjectOfSells[sale].itemId;
+				obj['price'] = jsonObjectOfSells[sale].salePrice;
+				obj['amount'] = jsonObjectOfSells[sale].amount;
+				obj['sellerId'] = jsonObjectOfSells[sale].sellerId;
+				obj['saleId'] = jsonObjectOfSells[sale]._id;
```

Comment:
It's awkward to access an object's membership field as object['stringID'] when you could much more easily do object.stringID

In general, this isn't the way of going about things. But in this instance, this shouldn't be a dictionary mapping strings to (somethings?), it should be its own object and type. And the logic for extracting it from the backend shouldn't be inside a component.